In [96]:
# Dependencies and Setup
import pandas as pd
import numpy as np

# File to Load 
file_to_load = "Resources/purchase_data.csv"

# Read Purchasing File and store into Pandas data frame
purchase_data = pd.read_csv(file_to_load)
purchase_data.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

## Player Count

* Display the total number of players


In [97]:
# Calculate total number of players
total_players = len(purchase_data['SN'].unique())
total_players_df = pd.DataFrame({'Total Players': [total_players]})
total_players_df

Total Players
0            576

## Purchasing Analysis (Total)

* Run basic calculations to obtain number of unique items, average price, etc.


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame


In [98]:
Items = purchase_data["Item Name"].nunique()
average_price = round(purchase_data["Price"].mean(), 2)
number_of_purchases = purchase_data["Item Name"].nunique()
total_revenue = purchase_data["Price"].sum()

purchase_summary_table = pd.DataFrame({"Number Unique Items": [Items],
                                      "Average Price": [average_price],
                                      "Number Of Purchases": [number_of_purchases],
                                      "Total Revenue": [total_revenue]
                                      })

purchase_summary_table['Average Price'] = purchase_summary_table['Average Price'].map('${:.2f}'.format)
purchase_summary_table['Total Revenue'] = purchase_summary_table['Total Revenue'].map('${:,.2f}'.format)

purchase_summary_table

Number Unique Items Average Price  Number Of Purchases Total Revenue
0                  179         $3.05                  179     $2,379.77

## Gender Demographics

* Percentage and Count of Male Players


* Percentage and Count of Female Players


* Percentage and Count of Other / Non-Disclosed




In [88]:
gender_group = purchase_data.groupby('Gender')
gender_sort = gender_group['SN'].nunique().sort_values(ascending=False)
gender_df = pd.DataFrame(gender_sort)


gender_df = gender_df.rename(columns={'SN': 'Total Count'})
gender_df['Percentage of Players'] = ((gender_df['Total Count'] / total_players) * 100).map('{:.2f}%'.format)
gender_df.index.name=None
gender_df

Total Count Percentage of Players
Male                           484                84.03%
Female                          81                14.06%
Other / Non-Disclosed           11                 1.91%


## Purchasing Analysis (Gender)

* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. by gender




* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [95]:
g_p = purchase_data.groupby(["Gender"]).sum()["Price"]
g_avg = purchase_data.groupby(["Gender"]).mean()["Price"]
g_counts = purchase_data.groupby(["Gender"]).count()["Price"]


Avg_p = g_p / purchase_data.groupby(["Gender"]).nunique()["SN"]

gender_data = pd.DataFrame({"Purchase Count": g_counts, 
                            "Average Purchase Price": g_avg.map("${:.2f}".format),
                            "Total Purchase Value": g_p.map("${:.2f}".format),
                            "Avg Total Purchase per Person": Avg_p.map("${:.2f}".format)})
gender_df.index.name=None
gender_data

Purchase Count Average Purchase Price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Avg Total Purchase per Person  
Gender                                                                    
Female                             $361.94                         $4.47  
Male                              $1967.64                         $4.07  
Other / Non-Disclosed               $50.19                         $4.56

## Age Demographics

* Establish bins for ages


* Categorize the existing players using the age bins. Hint: use pd.cut()


* Calculate the numbers and percentages by age group


* Create a summary data frame to hold the results


* Optional: round the percentage column to two decimal points


* Display Age Demographics Table


In [90]:
age_bins = [0, 9, 14, 19, 24, 29, 34, 39, 100]
group_names = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"]


purchase_data["Age Ranges"] = pd.cut(purchase_data["Age"], age_bins, labels=group_names)

age_demographics_totals = purchase_data.groupby(["Age Ranges"]).nunique()["SN"]
age_demographics_percents = age_demographics_totals / purchase_data['SN'].nunique() * 100

age_demographics = pd.DataFrame({"Total Count": age_demographics_totals, "Percent of Players": age_demographics_percents})

age_demographics

Total Count  Percent of Players
Age Ranges                                 
<10                  17            2.951389
10-14                22            3.819444
15-19               107           18.576389
20-24               258           44.791667
25-29                77           13.368056
30-34                52            9.027778
35-39                31            5.381944
40+                  12            2.083333

## Purchasing Analysis (Age)

* Bin the purchase_data data frame by age


* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. in the table below


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [91]:
item_count = purchase_data.groupby(["Item ID","Item Name"]).count()["Price"]
item_total = purchase_data.groupby(["Item ID","Item Name"]).sum()["Price"]
item_average = purchase_data.groupby(["Item ID","Item Name"]).mean()["Price"]

item_df = pd.DataFrame({"Purchase Count": item_count,
                          "Item Price": item_average.map("${:.2f}".format),
                          "Total Purchase Value": item_total.map("${:.2f}".format)})

item_df.sort_values("Purchase Count", ascending=False).head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
132     Persuasion                                        $3.22   
108     Extraction, Quickblade Of Trembling Hands         $3.53   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

## Top Spenders

* Run basic calculations to obtain the results in the table below


* Create a summary data frame to hold the results


* Sort the total purchase value column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [92]:
count_users = purchase_data.groupby(["SN"]).count()["Price"]
total_users = purchase_data.groupby(["SN"]).sum()["Price"]
average_users =round(purchase_data.groupby(["SN"]).mean()["Price"], 2)


user_df = pd.DataFrame({"Purchase Count": count_users,
                          "Total Purchase Amount": total_users,
                          "Average Purchase Price": average_users.map("${:.2f}".format)})

user_df.sort_values("Total Purchase Amount", ascending=False).head()

Purchase Count  Total Purchase Amount Average Purchase Price
SN                                                                       
Lisosia93                 5                  18.96                  $3.79
Idastidru52               4                  15.45                  $3.86
Chamjask73                3                  13.83                  $4.61
Iral74                    4                  13.62                  $3.40
Iskadarya95               3                  13.10                  $4.37

## Most Popular Items

* Retrieve the Item ID, Item Name, and Item Price columns


* Group by Item ID and Item Name. Perform calculations to obtain purchase count, average item price, and total purchase value


* Create a summary data frame to hold the results


* Sort the purchase count column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [93]:
item_count = purchase_data.groupby(["Item ID","Item Name"]).count()["Price"]
item_total = purchase_data.groupby(["Item ID","Item Name"]).sum()["Price"]
item_average = purchase_data.groupby(["Item ID","Item Name"]).mean()["Price"]

item_df = pd.DataFrame({"Purchase Count": item_count,
                          "Item Price": item_average.map("${:.2f}".format),
                          "Total Purchase Value": item_total.map("${:.2f}".format)})

item_df.sort_values("Purchase Count", ascending=False).head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
132     Persuasion                                        $3.22   
108     Extraction, Quickblade Of Trembling Hands         $3.53   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

## Most Profitable Items

* Sort the above table by total purchase value in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the data frame



In [94]:
item_count = purchase_data.groupby(["Item ID","Item Name"]).count()["Price"]
item_total = purchase_data.groupby(["Item ID","Item Name"]).sum()["Price"]
item_average = purchase_data.groupby(["Item ID","Item Name"]).mean()["Price"]

item_df = pd.DataFrame({"Purchase Count": item_count,
                          "Item Price": item_average.map("${:.2f}".format),
                          "Total Purchase Value": item_total})

item_df.sort_values("Total Purchase Value", ascending=False).head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
103     Singed Scalpel                                             8   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
82      Nirvana                                           $4.90   
145     Fiery Glass Crusader                              $4.58   
103     Singed Scalpel                                    $4.35   

                                                      Total Purchase Value  
Item ID Item Name                                                           
92      Final Critic                                                 59.99  
178     Oathbreaker, Last Hope of the Breaking Storm                 50.76  
82      Nirvana                                                      44.10  
145     Fiery Glass Crusader                                         41.22  
103     Singed Scalpel                                               34.80

Observable trends:

•	Gender purchase analysis:
84% of players are males (484) and only 14% of players are females (81). The average total purchase per person is almost the same ($4.47 for male and $4.07 for female)

•	Age demographics purchase analysis: 
44.79% of players are between 20 and 24 years old, followed by 18.57% between the age of 15 and 19 years old. The players between the age of 25 and 29 represent 13.36 of the players.

•	Most popular items and most profitable items:
The most popular item is Final Critic with 13 purchases followed by Oathbreaker, Last Hope of the Breaking Storm with 12 purchases. These 2 items are the most profitable too followed in the third place by Nirvana.
